src/client.rs

```diff
@@ -101,11 +111,94 @@ impl Client {
 
 // `bitcoind` RPC methods
 impl Client {
-    /// Get best block hash.
+    /// Get block
+    pub fn get_block(&self, block_hash: &BlockHash) -> Result<Block, Error> {
+        let hex_string: String = self.call("getblock", &[json!(block_hash), json!(0)])?;
+
+        let bytes: Vec<u8> = Vec::<u8>::from_hex(&hex_string).map_err(Error::HexToBytes)?;
+
+        let block: Block = deserialize(&bytes)
+            .map_err(|e| Error::InvalidResponse(format!("failed to deserialize block: {e}")))?;
+
+        Ok(block)
+    }
+
+    /// Get block verboseone
+    pub fn get_block_verbose(&self, block_hash: &BlockHash) -> Result<GetBlockVerboseOne, Error> {
+        let res: GetBlockVerboseOne = self.call("getblock", &[json!(block_hash), json!(1)])?;
+        Ok(res)
+    }
+
+    /// Get best block hash
     pub fn get_best_block_hash(&self) -> Result<BlockHash, Error> {
         let res: String = self.call("getbestblockhash", &[])?;
         Ok(res.parse()?)
     }
+
+    /// Get block count
+    pub fn get_block_count(&self) -> Result<u64, Error> {
+        let res: GetBlockCount = self.call("getblockcount", &[])?;
+        Ok(res.0)
+    }
+
+    /// Get block hash
+    pub fn get_block_hash(&self, height: u32) -> Result<BlockHash, Error> {
+        let raw: serde_json::Value = self.call("getblockhash", &[json!(height)])?;
+
+        let hash_str = match raw {
+            serde_json::Value::String(s) => s,
+            serde_json::Value::Object(obj) => obj
+                .get("hash")
+                .and_then(|v| v.as_str())
+                .ok_or_else(|| Error::InvalidResponse("getblockhash: missing 'hash' field".into()))?
+                .to_string(),
+            _ => {
+                return Err(Error::InvalidResponse(
+                    "getblockhash: unexpected response type".into(),
+                ));
+            }
+        };
+
+        BlockHash::from_str(&hash_str).map_err(Error::HexToArray)
+    }
+
+    /// Get block filter
+    pub fn get_block_filter(&self, block_hash: BlockHash) -> Result<GetBlockFilter, Error> {
```

Comment:
For consistency we should decide whether the API can take a block hash by value or by reference.

```suggestion
    pub fn get_block_filter(&self, block_hash: &BlockHash) -> Result<GetBlockFilter, Error> {
```